File: validParenthesis.js
var isValid = function(s) {
    openBrackets = ["(", "{", "["]
    parenthesisHash = {
        ")": "(",
        "}": "{",
        "]": "["
    }

    stack = []

    for (let i=0; i<s.length; i++){
        if (openBrackets.includes(s[i])){
            stack.push(s[i])
        } else if (!stack.length){
            return false
        } else if (stack[stack.length-1] !== parenthesisHash[s[i]]){
            return false
        } else {
            stack.pop()
        }
    }
    
    return stack.length === 0
};

console.log(isValid("()"))

File: valid_parenthesis.rb
require "byebug"

def is_valid(s)
    close_to_open  = {
        ")" => "(",
        "}" => "{",
        "]" => "["
    }

    bracket_stack = []

    s.each_char {|char|
        if close_to_open.keys().include?(char)
            if bracket_stack[-1] != close_to_open[char]
                return false
            else 
                bracket_stack.pop()
            end

        else
            bracket_stack.push(char)
        end
    }
    debugger
    bracket_stack.length == 0
    
end

print is_valid("()")